### ISSUE TITLE
persistence files for tests are not stored in "target"  directory

### ISSUE THREAD

**Issue by windbender** [moquette-io/moquette#288]
If the files are "cleaned" before use, this doesn't cause a huge problem.

However if the format of the files changes from build to build, then the file cannot even be properly opened by the program and so cannot be cleaned.  This leads to test failures.

```
checkPublishPassThroughCluster_Qos0(io.moquette.server.ServerIntegrationHazelcastHandlerInterceptorTest)  Time elapsed: 2.578 sec  <<< ERROR!
java.lang.RuntimeException: java.lang.ClassNotFoundException: io.moquette.parser.proto.messages.AbstractMessage$QOSType
	at org.mapdb.SerializerPojo.classForName(SerializerPojo.java:96)
	at org.mapdb.SerializerPojo$FieldInfo.<init>(SerializerPojo.java:200)
	at org.mapdb.SerializerPojo$1.deserialize(SerializerPojo.java:74)
	at org.mapdb.SerializerPojo$1.deserialize(SerializerPojo.java:39)
	at org.mapdb.Store.deserialize(Store.java:297)
```

Suggest storing these files within "target"  for the tests so they're removed during `mvn clean`

**Comment by andsel** [comment]
Hi @windbender,which version of Moquette are you using?The current master
doesn't have anymore the parser package
Il giorno 22/mar/2017 23:46, "windbender" <notifications@github.com> ha
scritto:

**Comment by windbender** [comment]
Yes precisely!   That package isn't used anymore, nor the classes in it.  The problem is that the previous builds in my directory DID use that class.  And then, after pulling in the latest code, when I ran the unit tests  the MabDB persistence failed because the classes governing the objects stored within were not longer available to "rehydrate" into.  The solution is to ensure that tests only use storage within the target directory.

**Comment by andsel** [comment]
Fixed by 9440352e565b09de6dc7170f8ea37f5fb8665ea9
